pkgs/applications/editors/vscode/extensions/README.md

```diff
@@ -0,0 +1,14 @@
+# Visual Studio Code Extensions
+
+## Conventions for adding new extensions
+
+* Extensions are named in the **lowercase** version of the extension's unique identifier. Which is found on the marketplace extension page, and is the name under which the extension is installed by VSCode under `~/.vscode`.
+  Extension location should be: ${lib.strings.toLower mktplcRef.publisher}.${lib.string.toLower mktplcRef.name}
+
+* Move extension to a discrete directory whenever the extension needs `extra parameters` and other files (such as `patches` or `components`). The parameters that exist in the global index file are utilities shared by many extensions. Parameters that are specific to an extension or that are not globally used shouldn't be added to the global index page.
```

Comment:
I would clarify what is meant by "extra parameters"; perhaps instead say "if it would require additional parameters to be added at the top of the file". I don't think it's *unambiguous* per se, since there's only really one place in the file where there are parameters, but the current wording leads me (and I imagine will lead other readers also) to think more about the extension expression itself, which will probably be situated partway through the file.

Reply:
I have reworded it. Feel free to provide further feedback.